extensions/security/deployment/src/main/java/io/quarkus/security/deployment/DotNames.java

```diff
@@ -7,11 +7,13 @@
 import org.jboss.jandex.DotName;
 
 import io.quarkus.security.Authenticated;
+import io.quarkus.security.PermissionsAllowed;
 
 public final class DotNames {
 
     public static final DotName ROLES_ALLOWED = DotName.createSimple(RolesAllowed.class.getName());
     public static final DotName AUTHENTICATED = DotName.createSimple(Authenticated.class.getName());
+    public static final DotName PERMISSIONS_ALLOWED = DotName.createSimple(PermissionsAllowed.class.getName());
     public static final DotName DENY_ALL = DotName.createSimple(DenyAll.class.getName());
     public static final DotName PERMIT_ALL = DotName.createSimple(PermitAll.class.getName());
```

Comment:
I noticed Permit All here, and wonder (not in relation to this file update :-)), what happens if we have
`@PermitAll` and `@PermissionsAllowed("a")` ?

Reply:
The rules of combination are exactly same as with other annotations. I actually had test that combined these exact annotation in RR and I dropped it as I didn't see a point. I can explain you why - how it works comes from a mechanism, not from my work. In short:

- when collecting, method level annotations are always picked before class level annotations (that's how it works for all security annotations - rolesallowed, authetnicated, deny all, permitall, permissionsallowed.)
- class level annotations are not picked when method level annotations exist
- method level annotations can't be combined - validation exception is thrown
- so you always have just one annotation and in case above-mentioned somehow (??) failed, `io.quarkus.security.runtime.interceptor.SecurityHandler` never runs more than once for one invoked method (`InvocationContext`)

I can add tests that proves it if you like.

Reply:
@sberyozkin 
I added `PermitAllWithPermissionsAllowedValidationFailureTest,RolesAllowedWithPermissionsAllowedValidationFailureTest,RolesAllowedWithPermissionsAllowedTest`. Please check them out for expected behavior.